src/query/block/scalar.go

```diff
@@ -85,11 +91,18 @@ func (b *Scalar) SeriesIter() (SeriesIter, error) {
 func (b *Scalar) Close() error { return nil }
 
 // Value returns the value for the scalar block
-func (b *Scalar) Value() float64 { return b.val }
+func (b *Scalar) Value(idx int) float64 {
+	var t time.Time
+	if idx != -1 {
```

Comment:
This actually should never be the case so not sure if it's worth handling the error

Reply:
no need, it's fine if we panic on a bug here. Also, don't have the if block, and just assume idx >=0